api/endpoints.go

```diff
@@ -132,6 +132,89 @@ func handleAccountLogout(w http.ResponseWriter, r *http.Request) {
 	w.WriteHeader(http.StatusOK)
 }
 
+func handleAddFriend(w http.ResponseWriter, r *http.Request) {
+	formErr := r.ParseForm()
+	if formErr != nil {
+		httpError(w, r, fmt.Errorf("failed to parse request form: %s", formErr), http.StatusBadRequest)
+		return
+	}
+
+	uuid, err := uuidFromRequest(r)
+	if err != nil {
+		httpError(w, r, err, http.StatusBadRequest)
+		return
+	}
+
+	success, err := db.AddFriend(uuid, r.Form.Get("username"))
+	message := "Success"
+	if !success {
+		message = err.Error()
```

Comment:
why just not return an error like 400?

Reply:
Getting a `user not exist` / `already friend` is not a really bad request, nor any other http error, the request & the application works correctly, it's a higher level error that gets handled on the client
Having 500 when there's a `... please contact an administrator ...` would be nice but that would be adding some code to handle this specific error that I don't feel is worth having for such a rare occasion